Geometry/HGCalCommonData/src/HGCalGeomRotation.cc

```diff
@@ -0,0 +1,196 @@
+#include "Geometry/HGCalCommonData/interface/HGCalGeomRotation.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+void HGCalGeomRotation::uvMappingFromSector0(WaferCentring waferCentring,
+                                             int& moduleU,
+                                             int& moduleV,
+                                             unsigned sector) const {
+  if (sector == 0) {
+    return;
+  }
+
+  if (_sectorType == SectorType::Sector60Degrees) {
+    uvMappingFrom60DegreeSector0(waferCentring, moduleU, moduleV, sector);
+  } else if (_sectorType == SectorType::Sector120Degrees) {
+    uvMappingFrom120DegreeSector0(waferCentring, moduleU, moduleV, sector);
+  }
+}
+
+unsigned HGCalGeomRotation::uvMappingToSector0(WaferCentring waferCentring, int& moduleU, int& moduleV) const {
+  unsigned sector = 0;
+
+  if (_sectorType == SectorType::Sector60Degrees) {
+    sector = uvMappingTo60DegreeSector0(waferCentring, moduleU, moduleV);
+  } else if (_sectorType == SectorType::Sector120Degrees) {
+    sector = uvMappingTo120DegreeSector0(waferCentring, moduleU, moduleV);
+  }
+
+  return sector;
+}
+
+void HGCalGeomRotation::uvMappingFrom60DegreeSector0(WaferCentring waferCentring,
+                                                     int& moduleU,
+                                                     int& moduleV,
+                                                     unsigned sector) const {
+  if (waferCentring != WaferCentring::WaferCentred) {
+    edm::LogError("HGCalGeomRotation")
+        << "HGCalGeomRotation: 60 degree sector defintion selected, but not WaferCentred centring. This is "
+           "incompatible, switching to WaferCentred centring";
+    waferCentring = WaferCentring::WaferCentred;
+  }
+
+  if (sector > 5) {
+    throw cms::Exception("RotationException")
+        << "HGCalGeomRotation: desired sector must be either 0, 1, 2, 3, 4, 5 or 6";
+  }
+  for (unsigned rot = 0; rot < sector; rot++) {
+    RotateModule60DegreesAnticlockwise(moduleU, moduleV);
+  }
+}
+
+void HGCalGeomRotation::uvMappingFrom120DegreeSector0(WaferCentring waferCentring,
+                                                      int& moduleU,
+                                                      int& moduleV,
+                                                      unsigned sector) const {
+  int offset;
+
+  if (waferCentring == WaferCentring::WaferCentred) {
+    offset = 0;
+  } else if (waferCentring == WaferCentring::CornerCentredY) {
+    offset = -1;
+  } else if (waferCentring == WaferCentring::CornerCentredMercedes) {
+    offset = 1;
+  } else {
+    throw cms::Exception("RotationException")
+        << "HGCalGeomRotation: WaferCentring must be one of: WaferCentred, CornerCentredY or CornerCentredMercedes";
+  }
+
+  if (sector > 2) {
+    throw cms::Exception("RotationException") << "HGCalGeomRotation: desired sector must be either 0, 1 or 2";
```

Comment:
I see that in other cases, `sector` = 3 is valid. That makes more likely that some bug would pass a value of 3 or more for `sector` to this method. I think this exception should be a `LogError` and `return`. That would result in misplaced modules, but at least the workflow could keep running.

Reply:
Ok sure, I've switched to `LogError` in this case.